src/pybind/mgr/volumes/fs/operations/versions/metadata_manager.py

```diff
@@ -75,15 +75,18 @@ def flush(self):
 
         fd = None
         try:
-            fd = self.fs.open(self.config_path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
+            tmp_config_path = self.config_path + b'.tmp'
+            fd = self.fs.open(tmp_config_path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
```

Comment:
Could attempting to create a temporary file fail as well ?
Is it possible to open the desired path (`self.config_path`) and then use `self.fs.fallocate(fd, ...)` to ensure availability of space an then write to it directly ?
You might have to explicitly truncate the `self.config_path` to the new size anyway.

Reply:
I meant `fallocate` was tried here [1] but configparser wasn't friendly with it. 

[1] https://github.com/ceph/ceph/pull/44347#discussion_r827822449